File: backend/Application/Transacoes/ListTransacoes/ListTransacoesRequest.cs
using System.ComponentModel.DataAnnotations;
using FinancialControllerServer.Domain.Enums;

namespace FinancialControllerServer.Application.Transacoes.ListTransacoes;

public class ListTransacoesRequest
{
    public DateOnly? DataInicio { get; set; }
    public DateOnly? DataFim { get; set; }

    [EnumDataType(typeof(TipoTransacao), ErrorMessage = "Tipo inválido")]
    public TipoTransacao? Tipo { get; set;}
    public Guid? PessoaId { get; set; }
    public Guid? CategoriaId { get; set; }
}

File: frontend/src/features/transacoes/types/listTransacoesRequest.ts
import type { TipoTransacao } from "../../../domain/enums/TipoTransacao";

export interface ListTransacoesRequest {
    dataInicio?:string;
    dataFim?: string;
    tipo?: TipoTransacao;
    pessoaId?: string;
    categoriaId?: string;
}